src/dsl.ts

```diff
@@ -328,14 +337,39 @@ export class SnykApiCheckDsl implements ApiCheckDsl {
       ...genericEntityRuleImpl<
         OpenApiResponseFact,
         ConceptualLocation,
-        SynkApiCheckContext,
+        SynkApiCheckContext & {
+          isSingletonPath: boolean;
+          requestDataPropertyIsArray: boolean;
+        },
         OpenAPIV3.ResponsesObject
       >(
         OpenApiKind.Response,
         dsl.changelog,
         dsl.nextFacts,
         (response) => `${response.statusCode}`,
-        (location) => dsl.getContext(location),
+        (location) => {
+          const dataPath = [
+            ...jsonPointerHelpers.decode(location.jsonPath).slice(0, 3),
```

Comment:
Yep -- I think we may add more first class support in a future DSL for rules that run for certain kinds of operations. It seems like all the complexity in our rule handlers is because of this indirection. We find out that 'x changed' and then we try to figure out the context of 'x' and apply a rule.